### DISCUSSION TITLE
Move "heads-up" preference change warning to the bottom of the dialog?

### DISCUSSION THREAD

**Discussion by jandubois** [rancher-sandbox/rancher-desktop#3264] [Archive]
When the user changes preferences that require resetting or restarting kubernetes (or the whole VM), we present a little warning overlay to alert the user to the fact that pressing "Apply" will initiate a lengthy action:

<img width="880" alt="Screen Shot 2022-10-24 at 9 54 07 AM" src="https://user-images.githubusercontent.com/78947/197584188-9e440b07-7d75-41d6-a08b-7292dddd653f.png">

This overlay remains visible regardless which page/tab is currently selected in the dialog.

We don't have any pages that use the full height of the page, so I don't know what happens when there is an overlap.

I always expected these messages to be in the bottom part, just left of the Cancel/Apply buttons because that's what it is closely related to: "If you press Apply, we will have to restart Kubernetes":

<img width="880" alt="Screen Shot 2022-10-24 at 9 59 59 AM" src="https://user-images.githubusercontent.com/78947/197585231-a5126cae-2ed7-43d9-bf4c-352a95382c61.png">

Edited screenshot just to show the area it should be displayed, actual positioning inside the box, and formatting (coloured background, differently coloured bar to the left) still to be discussed.

**Comment by mook-as** [comment]
We _do_ have pages that can use the whole height, on Windows: the list of WSL integrations can be unbounded. But that just means the top part scrolls, and the button bar is still visible all the time (otherwise you couldn't apply the changes…).

I'd probably move the warning bar to be left-aligned inside the button bar, rather than right-aligned, but in general that seems fine. Maybe reword it because it's not really _**Kubernetes** will restart…_ since that can apply to non-Kube things too; it's a leftover from when we only did Kube. (I thought we had an issue for this but couldn't find it at a quick glance.)

**Reply by rak-phillip** [reply]
FYI this was our original implementation for the banners in Preferences but it was requested to change after a UX review. See #2774

**Reply by jandubois** [reply]
> FYI this was our original implementation for the banners in Preferences but it was requested to change after a UX review. See #2774

I don't think I was involved in that UX review (I was on vacation most of August), and I think I disagree with the decision. I think the original implementation makes more sense. I'm curious about the arguments that justified this change.

> I'd probably move the warning bar to be left-aligned inside the button bar, 

I have no particular opinion on how it should be presented in the button bar, but we may want to have a "help" button in the bottom left corner in the future... Will file another discussion item about this soon.

**Comment by kwwii** [comment]
I think we should follow standard conventions and display the message closer to the UI responsible for the state. It's also more noticeable when clicking the Apply button.

**Comment by jandubois** [comment]
This has been implemented in the 1.7 release (issue #3427). Moving discussion to archive.
